src/libraries/Native/Unix/System.Globalization.Native/CMakeLists.txt

```diff
@@ -59,22 +59,27 @@ set(NATIVEGLOBALIZATION_SOURCES
 
 include_directories(${UTYPES_H})
 
-add_library(System.Globalization.Native
-    SHARED
-    ${NATIVEGLOBALIZATION_SOURCES}
-    ${VERSION_FILE_PATH}
-)
+if (CORECLR_BUILD)
+ include_directories("../Common")
+endif()
 
-target_link_libraries(System.Globalization.Native
-    dl
-)
+if (GEN_SHARED_LIB)
+    add_library(System.Globalization.Native
+        SHARED
+        ${NATIVEGLOBALIZATION_SOURCES}
+        ${VERSION_FILE_PATH}
+    )
 
-install_library_and_symbols (System.Globalization.Native)
+    target_link_libraries(System.Globalization.Native
+        dl
+    )
+
+    install_library_and_symbols (System.Globalization.Native)
+endif()
 
 add_library(System.Globalization.Native-Static
```

Comment:
I wonder why do we generate the static library when building libraries. Was it ever used by anything? If not, then I'd prefer naming the static library the same as the dynamic one. The coreclr build is composed of many static libraries and none of them have a suffix indicating it is static.

Reply:
The static builds of the shims are used by CoreRT and Mono/WebAssembly. Mono/Webassembly does not use System.Globalization currently, but it does use some of the other ones.

> suffix indicating it is static.

IIRC, this suffix was added to make cmake(?) happy.

Reply:
I'm not sure if it was used before, we do that for all the native libraries we build in libraries. I believe for wasm we need to build them, but when building wasm, we don't build this library nor the shared libraries. I can update this to only build when `NOT GEN_SHARED_LIB`.

Reply:
I'll leave as is given that we might still need it for Mono/WebAssembly in the future. When mono links to this library we can stop building the SHARED library at all and then remove the -Static suffix.

Reply:
The `-Static` suffix (or any other way to differentiate the static and dynamic libraries) is needed when both dynamic and static versions are produced. If we were producing just one, it would not be needed. I was referring to the case when we would be building just one flavor.

Reply:
Yeah, I understand that. When building libraries we build both as of now. Do we want to stop building the Static as part of libraries build?

Reply:
Based on the discussion above, let's keep it.